OpenXMLSDK.Engine/Word/ReportEngine/Renders/ForEachExtensions.cs

```diff
@@ -48,6 +47,16 @@ public static class ForEachExtensions
                     // Specific rule for TemplateModels in childs :
                     if (clone is TemplateModel)
                     {
+                        // Search the template ID to use
+                        var templateIdToFindInCurrentContext = (clone as TemplateModel).TemplateId;
```

Comment:
Note :
You can replace the code by a context.ReplateItem(clone as TemplateModel), and create a ReplateItem extension for type TemplateModel. The ReplaceItem extension will fill the TemplateId field.

Reply:
I add a method to replace the item in the TemplateModelExtension.